app/helpers/samples_helper.rb

```diff
@@ -30,21 +30,17 @@ def controlled_vocab_form_field(sample_controlled_vocab, element_name, values, l
     end
   end
 
-  def list_select_form_field(attribute,element_name, values)
+  def controlled_vocab_list_form_field(sample_controlled_vocab, element_name, values)
 
-    sample_controlled_vocab =  attribute.sample_controlled_vocab
-
-    options = options_from_collection_for_select(
-        sample_controlled_vocab.sample_controlled_vocab_terms.sort_by(&:label),
-        :label, :label,
-        values
-      )
-      select_tag element_name,
-                 options,
-                 class: "form-control cv_list_attribute",
-                 include_blank: "",
-                 name: "#{element_name}[]",
-                 multiple: "multiple"
+    scv_id = sample_controlled_vocab.id
+    is_ontology = sample_controlled_vocab.source_ontology.present?
+    existing_objects = Array(values).collect do |value|
+      Struct.new(:id, :name).new(value, value)
```

Comment:
The `Struct` should be moved outside the loop (and maybe put in a constant), or it will create a new type of Struct for each iteration